<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1057</width>
    <height>349</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Remote access and publishing</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabPosition">
       <enum>QTabWidget::West</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>PubNub</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QFrame" name="frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <property name="class" stdset="0">
           <string>PeripheralCollection</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QFrame" name="frame_2">
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <property name="class" stdset="0">
              <string>PeripheralCollectionInner</string>
             </property>
             <layout class="QGridLayout" name="gridLayout">
              <property name="topMargin">
               <number>2</number>
              </property>
              <property name="rightMargin">
               <number>2</number>
              </property>
              <property name="bottomMargin">
               <number>2</number>
              </property>
              <property name="spacing">
               <number>2</number>
              </property>
              <item row="2" column="1">
               <widget class="QLabel" name="channelLabel">
                <property name="text">
                 <string>channel:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QCheckBox" name="checkBox">
                <property name="text">
                 <string>Listen to incoming messages</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLineEdit" name="pubEdit">
                <property name="text">
                 <string/>
                </property>
                <property name="placeholderText">
                 <string>publish key</string>
                </property>
               </widget>
              </item>
              <item row="1" column="2">
               <widget class="QPushButton" name="pushButton_4">
                <property name="text">
                 <string>Reset keys</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLineEdit" name="subEdit">
                <property name="text">
                 <string/>
                </property>
                <property name="placeholderText">
                 <string>subscriber key</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QFrame" name="frame_3">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <property name="class" stdset="0">
              <string>PeripheralCollectionInner</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_2">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <property name="spacing">
               <number>0</number>
              </property>
              <item row="1" column="0">
               <widget class="QFrame" name="frame_4">
                <property name="frameShape">
                 <enum>QFrame::StyledPanel</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Raised</enum>
                </property>
                <property name="class" stdset="0">
                 <string>PeripheralCollectionInner</string>
                </property>
                <layout class="QGridLayout" name="gridLayout_3">
                 <item row="0" column="0">
                  <widget class="QLineEdit" name="cmdEdit">
                   <property name="text">
                    <string>get_version()</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="2">
                  <widget class="QLineEdit" name="sendID">
                   <property name="maximumSize">
                    <size>
                     <width>60</width>
                     <height>16777215</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>0x02</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0" colspan="3">
                  <widget class="QLabel" name="responseLabel">
                   <property name="text">
                    <string>Response:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QPushButton" name="pushButton_3">
                   <property name="text">
                    <string>Execute on</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QTextEdit" name="results">
                <property name="textInteractionFlags">
                 <set>Qt::NoTextInteraction</set>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>ThingSpeak</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QFrame" name="frame_5">
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <property name="class" stdset="0">
           <string>PeripheralCollectionInner</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <property name="topMargin">
            <number>2</number>
           </property>
           <property name="rightMargin">
            <number>2</number>
           </property>
           <property name="bottomMargin">
            <number>2</number>
           </property>
           <property name="spacing">
            <number>2</number>
           </property>
           <item row="1" column="1">
            <widget class="QLineEdit" name="thingSpeakKey">
             <property name="toolTip">
              <string>Get a key from thingspeak.com by creating your own
                                                            channel
                                                        </string>
             </property>
             <property name="text">
              <string>H1OE0R294J0R6YQN</string>
             </property>
             <property name="placeholderText">
              <string>publish key</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>ThingSpeak Key</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QTextBrowser" name="results_2">
          <property name="textInteractionFlags">
           <set>Qt::TextEditorInteraction</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="frame_6">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <property name="class" stdset="0">
           <string>PeripheralCollectionInner</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="0" column="4">
            <widget class="QSpinBox" name="thingSpeakInterval">
             <property name="toolTip">
              <string>Set the interval for pushing data.
                                                            The rate limit for ThingSpeak.com is 15 seconds
                                                        </string>
             </property>
             <property name="suffix">
              <string>Seconds</string>
             </property>
             <property name="minimum">
              <number>15</number>
             </property>
             <property name="maximum">
              <number>1000</number>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="pushButton_6">
             <property name="text">
              <string>Set Command</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLineEdit" name="cmdEditThingSpeak">
             <property name="text">
              <string>get_average_voltage('CH1')</string>
             </property>
             <property name="placeholderText">
              <string>Command to execute</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Update Interval</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>checkBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>setListenState(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>220</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>198</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>resetKeys()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>963</x>
     <y>31</y>
    </hint>
    <hint type="destinationlabel">
     <x>821</x>
     <y>14</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>execRemote()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>883</x>
     <y>311</y>
    </hint>
    <hint type="destinationlabel">
     <x>821</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>setThingSpeakCommand()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>130</x>
     <y>51</y>
    </hint>
    <hint type="destinationlabel">
     <x>331</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setListenState(bool)</slot>
  <slot>execRemote()</slot>
  <slot>resetKeys()</slot>
  <slot>setThingSpeakCommand()</slot>
  <slot>setThingSpeakKey()</slot>
 </slots>
</ui>
